lib/public_include/fty_shm.h

```diff
@@ -77,9 +88,21 @@ class shmMetrics
     // If you use this, DO NOT DELETE the fty_proto_t. It will be take
     // care by the shmlMetrics's destructor.
     //  (same warning if you access to it using iterator)
+
+    // @param index : index of the m_metricsVector
+    // @return : the address of the indexed element
     fty_proto_t*      get(int index);
+
+    // duplicates the index element
+    // @param index : index of the m_metricsVector
+    // @return : duplicated element
     fty_proto_t*      getDup(int index);
+
+    // pushes back the input fty_proto_t pointer
+    // @param metric : member will be pushed back to the vector
```

Comment:
is the pointer reference strored inside the object or is it a copy... what if i delete the proto* on return?